[CQT-345] Add changelog to cQASM-spec (#86)
QuTech-Delft/cQASM-spec: CHANGELOG.md

# Changelog

All notable changes to this project will be documented in this file.

### Types of changes:
* **Added** for new features.
* **Changed** for changes in existing functionality.
* **Fixed** for any bug fixes.
* **Removed** for now removed features.

> ## [3.0-beta3] - [ xxxx-yy-zz ]
> 
> 
> 

Reviewer: ```suggestion
```

Should we better let the first feature PR to add this section?

Author: It is there as a placeholder so that, as features are added, they can be recorded under this header. Of course the exact version may be dependent on the changes that are made in, which will be determined at the moment of the release. So we could right instead

```suggestion
## [<version>] - [ xxxx-yy-zz ]

```

or something similar.

Reviewer: OK!

## [3.0-beta2] - [ 2025-02-18 ]

### Added

> - `init` non-unitary instruction
> - `SWAP` two-qubit unitary instruction
> - `barrier` and `wait` control instructions
> - Gate modifiers
>   - inverse `inv`
>   - power `pow()`
>   - control `ctrl`

Reviewer: ```suggestion
- `SWAP` two-qubit unitary instruction.
- `init` non-unitary instruction.
- `barrier` and `wait` control instructions.
- `inv`, `pow`, and `ctrl` gate modifiers.
```

Reorder text just trying to follow a logic: unitary, non-unitary, and control instructions, and gate  modifiers.

Wouldn't it be clearer ending sentences with a period?

## [3.0-beta1] - [ 2024-10-16 ]

The first (beta) release of the new cQASM language.

Main changes with respect to previous cQASM versions are:

> - Case sensitive syntax
> - Decoupled qubit and bit registers
> - Qubit and bit register variable declarations
> - `q` and `b` are no longer reserved characters
> - Assignment of measurement outcomes to bit variables
> - No component (e.g. translator/compiler/simulator) specific directives (for now)
> - Single-line comments through double-slash notation (`//`), instead of the hashtag notation (`#`)

Reviewer: ```suggestion
- Case sensitive syntax.
- Decoupled qubit and bit registers.
- Qubit and bit register variable declarations.
- `q` and `b` are no longer reserved identifiers.
- Assignment of measurement outcomes to bit variables.
- Single-line comments through double-slash notation (`//`), instead of the hashtag notation (`#`).
```

> For a full description of the new language we refer to its [online specification](https://qutech-delft.github.io/cQASM-spec/).
> 
> > **Note**: that this version of cQASM is not backwards compatible with previous versions of the language.

Reviewer: ```suggestion
> **Note**: this version of cQASM is not backwards compatible with previous versions of the language.
```

I wouldn't add a link to the documentation in the changelog.

Author: Yes, good point!
